src/models/filter.rs

```diff
@@ -67,98 +67,99 @@ impl Constraint {
-  (matcher = $matcher:expr, queries= [$($q:expr,)*]) => {
-    $crate::models::constraint::ConstraintBuilder::default()
-      .matcher($crate::utilities::tree_sitter_utilities::TSQuery::new($matcher.to_string()))
-      .queries(vec![$($crate::utilities::tree_sitter_utilities::TSQuery::new($q.to_string()),)*])
+macro_rules! filter {
+  (enclosing_node = $enclosing_node:expr, not_contains= [$($q:expr,)*]) => {
+    $crate::models::filter::FilterBuilder::default()
+      .enclosing_node($crate::utilities::tree_sitter_utilities::TSQuery::new($enclosing_node.to_string()))
+      .not_contains(vec![$($crate::utilities::tree_sitter_utilities::TSQuery::new($q.to_string()),)*])
       .build().unwrap()
   };
 }
 
-pub use constraint;
+pub use filter;
 
-impl Instantiate for Constraint {
+impl Instantiate for Filter {
   /// Create a new query from `self` by updating the `query` and `replace` based on the substitutions.
-  fn instantiate(&self, substitutions_for_holes: &HashMap<String, String>) -> Constraint {
-    Constraint {
-      matcher: self.matcher().instantiate(substitutions_for_holes),
-      queries: self
-        .queries()
+  fn instantiate(&self, substitutions_for_holes: &HashMap<String, String>) -> Filter {
+    Filter {
+      enclosing_node: self.enclosing_node().instantiate(substitutions_for_holes),
+      not_contains: self
+        .not_contains()
         .iter()
         .map(|x| x.instantiate(substitutions_for_holes))
         .collect_vec(),
     }
   }
 }
 
-// Implements instance methods related to applying a constraint
+// Implements instance methods related to applying a filter
 impl SourceCodeUnit {
   pub(crate) fn is_satisfied(
     &self, node: Node, rule: &InstantiatedRule, substitutions: &HashMap<String, String>,
     rule_store: &mut RuleStore,
   ) -> bool {
     let mut updated_substitutions = self.piranha_arguments().input_substitutions();
     updated_substitutions.extend(substitutions.clone());
-    rule.constraints().iter().all(|constraint| {
-      self._is_satisfied(constraint.clone(), node, rule_store, &updated_substitutions)
-    })
+    rule
+      .filters()
+      .iter()
+      .all(|filter| self._is_satisfied(filter.clone(), node, rule_store, &updated_substitutions))
   }
 
-  /// Checks if the node satisfies the constraints.
-  /// Constraint has two parts (i) `constraint.matcher` (ii) `constraint.query`.
-  /// This function traverses the ancestors of the given `node` until `constraint.matcher` matches
-  /// i.e. finds scope for constraint.
-  /// Within this scope it checks if the `constraint.query` DOES NOT MATCH any sub-tree.
+  /// Checks if the node satisfies the filters.
+  /// filter has two parts (i) `filter.enclosing_node` (ii) `filter.query`.
+  /// This function traverses the ancestors of the given `node` until `filter.enclosing_node` matches
+  /// i.e. finds scope for filter.
+  /// Within this scope it checks if the `filter.query` DOES NOT MATCH any sub-tree.
   fn _is_satisfied(
```

Comment:
should we rename `_is_satisfied` to `test` or `check` ? So that it is in spirit of `filters` and not `constraints` ?

Reply:
`check` is good